### DISCUSSION TITLE
How to make Gitlab CI interpret the "gitlab" output format of Spectral CLI ?

### DISCUSSION THREAD

**Discussion by pidj** [stoplightio/spectral#2761] [Q&A]
Hi folks,
I have a question, for whoever would have the answer and be kind enough to reply :-)

I use the Spectral CLI to lint OpenAPI/AsyncAPI files. In the available output formats, one of them is "gitlab", which generates a report in JSON. Given the name of that formatter, I expect it's meant at being interpreted by Gitlab CI/CD in some way (e.g. for a MR report), but I can't figure out how.
My first guess was to feed this as a "report artifact" of a Gitlab job, but if it's correct, which one among the list provided by Gitlab (https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html) ?

Thank you in advance if anyone of you know what to do with this JSON output of the "gitlab" formatter ! :-)

**Comment by frankkilcommins** [comment]
The PR that add this functionality was focused around [Code Climate](https://gitlab.com/gitlab-org/codeclimate). 

@smsalisbury can you comment if the format will work more generally within GitLab CI or if it's limited to codeclimate type reports?

**Reply by smsalisbury** [reply]
Yes, GitLab uses a subset of the CodeClimate spec for its `codequality` report.

https://docs.gitlab.com/ee/ci/testing/code_quality.html#code-quality-report-format
https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportscodequality

**Reply by pidj** [reply]
Thank you ! The issue must be on my side then, I will continue to try and make it work. Sorry for bothering.
